kvserver: remove StoreBenignError

tbg opened the pull request:
Before commit 3f0b37af29fd4eb15599c44d73dc237dd33e0b1a, the StoreBenignError is used to handle pebble.ErrSnapshotExcised. As the latter has been removed from pebble, we don't need StoreBenignError anymore.

This commit does the following:

- Remove type "StoreBenignError".
- Remove the related increase action on counter "storeFailures".
- Update related tests "TestBaseQueueRequeue".

Fixes: #129941

Closes: #130308

Release note: None

XiaochenCui committed:
kvserver: remove StoreBenignError

Before commit 3f0b37af29fd4eb15599c44d73dc237dd33e0b1a, the
StoreBenignError is used to handle pebble.ErrSnapshotExcised. As the
latter has been removed from pebble, we don't need StoreBenignError
anymore.

This commit does the following:

- Remove type "StoreBenignError".
- Remove the related increase action on counter "storeFailures".
- Update related tests "TestBaseQueueRequeue".

Fixes: #129941

Release note: None

stevendanna commented on pkg/kv/kvserver/queue.go:
```diff
@@ -1170,17 +1170,12 @@ func (bq *baseQueue) finishProcessingReplica(
 		// (and production), move benign failures into a dedicated category.
 		bq.failures.Inc(1)
-		if storeBenign {
-			bq.storeFailures.Inc(1)
```
I am not sure this is incremented anywhere else so we could potentially remove it?

tbg replied:
oh, you're right! PTAL at new commit.

tbg committed:
kvserver: remove unused storage.queue.store-failures metric

This was never incremented.

Epic: none

Release note (ops change): the `storage.queue.store-failures` metric was removed.

tbg commented:
TFTR!

bors r+
(fixed ./dev gen)
